tests/Support/SupportCollectionTest.php

```diff
@@ -1984,6 +1984,128 @@ public function testSortDesc($collection)
         $this->assertEquals(['T10', 'T2', 'T1'], $data->values()->all());
     }
 
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testSortRecursive($collection): void
+    {
+        $data = new $collection([
+            'users' => [
+                [
+                    // should sort associative arrays by keys
+                    'name' => 'joe',
+                    'mail' => 'joe@example.com',
+                    // should sort deeply nested arrays
+                    'numbers' => [2, 1, 0],
+                ],
+                [
+                    'name' => 'jane',
+                    'age' => 25,
+                ],
+            ],
+            'repositories' => [
+                // should use weird `sort()` behavior on arrays of arrays
+                ['id' => 1],
+                ['id' => 0],
+            ],
+            // should sort non-associative arrays by value
+            20 => [2, 1, 0],
+            30 => [
+                // should sort non-incrementing numerical keys by keys
+                2 => 'a',
+                1 => 'b',
+                0 => 'c',
+            ],
+        ]);
+
+        $expect = new $collection([
+            20 => [0, 1, 2],
+            30 => [
+                0 => 'c',
+                1 => 'b',
+                2 => 'a',
+            ],
+            'repositories' => [
+                ['id' => 0],
+                ['id' => 1],
+            ],
+            'users' => [
+                [
+                    'age' => 25,
+                    'name' => 'jane',
+                ],
+                [
+                    'mail' => 'joe@example.com',
+                    'name' => 'joe',
+                    'numbers' => [0, 1, 2],
+                ],
+            ],
+        ]);
+
+        $this->assertEquals($expect, $data->sortRecursive());
+    }
+
+    /**
+     * @dataProvider collectionClassProvider
+     */
```

Comment:
Use attribute instead of annotation for future compatibility.

```php
use PHPUnit\Framework\Attributes\DataProvider
```

```diff
-   /**
-    * @dataProvider collectionClassProvider
-    */
+   #[DataProvider('collectionClassProvider')
```